x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -239,15 +239,18 @@ protected LogicalPlan rule(Aggregate aggregate) {
                     // project away transient fields and re-enforce the original order using references (not copies) to the original aggs
                     // this works since the replaced aliases have their nameId copied to avoid having to update all references (which has
                     // a cascading effect)
-                    plan = new EsqlProject(source, plan, Expressions.asAttributes(aggs));
+                    plan = new Project(source, plan, Expressions.asAttributes(aggs));
```

Comment:
Small tweak - no need to return EsqlProject instead Project since the tree is already resolved.